File: codegen/results/v002/qwen3.6_35b_think/20260418_184802/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const REFERENCE_DATE = new Date(2025, 6, 1); // July 1, 2025

function calculateAge(birthdayStr: string): number {
  const [mStr, dStr, yStr] = birthdayStr.split('/');
  const month = parseInt(mStr, 10);
  const day = parseInt(dStr, 10);
  const year = parseInt(yStr, 10);
  const birthdayThisYear = new Date(REFERENCE_DATE.getFullYear(), month - 1, day);
  let age = REFERENCE_DATE.getFullYear() - year;
  if (birthdayThisYear > REFERENCE_DATE) {
    age--;
  }
  return age;
}

function formatBirthday(birthdayStr: string): string {
  const [m, d, y] = birthdayStr.split('/');
  return `${y}-${m.padStart(2, '0')}-${d.padStart(2, '0')}`;
}

function splitName(name: string): { firstName: string; lastName: string } {
  const parts = name.trim().split(/\s+/);
  return {
    firstName: parts[0],
    lastName: parts[parts.length - 1],
  };
}

function splitRelative(relative: string): { firstName: string; lastName: string } | null {
  if (!relative || relative === 'null') return null;
  const parts = relative.trim().split(/\s+/);
  return { firstName: parts[0], lastName: parts[parts.length - 1] };
}

const csvContent = fs.readFileSync(path.join('input', 'input.csv'), 'utf-8');
const lines = csvContent.trim().split('\n');
const headers = lines[0].split(',');

const result = lines.slice(1).map((line) => {
  const values = line.split(',');
  const row: Record<string, string> = {};
  headers.forEach((h, i) => {
    row[h.trim()] = values[i] ? values[i].trim() : '';
  });

  const { firstName, lastName } = splitName(row.Name);
  const birthday = formatBirthday(row.Birthday);
  const age = calculateAge(row.Birthday);

  const relatives: Array<{ FirstName: string; LastName: string; Relationship: string }> = [];
  const relationshipMap: Array<[string, string]> = [
    ['Father', 'Father'],
    ['Mother', 'Mother'],
    ['Brother', 'Brother'],
    ['Sister', 'Sister'],
  ];

  for (const [col, rel] of relationshipMap) {
    const parsed = splitRelative(row[col]);
    if (parsed) {
      relatives.push({ FirstName: parsed.firstName, LastName: parsed.lastName, Relationship: rel });
    }
  }

  return { FirstName: firstName, LastName: lastName, Birthday: birthday, Age: age, Relatives: relatives };
});

console.log(JSON.stringify(result, null, 2));

File: codegen/results/v002/qwen3.6_35b_think/20260418_184802/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strings"
	"time"
)

type Relative struct {
	FirstName    string `json:"FirstName"`
	LastName     string `json:"LastName"`
	Relationship string `json:"Relationship"`
}

type Person struct {
	FirstName string     `json:"FirstName"`
	LastName  string     `json:"LastName"`
	Birthday  string     `json:"Birthday"`
	Age       int        `json:"Age"`
	Relatives []Relative `json:"Relatives"`
}

func main() {
	f, err := os.Open("input/input.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer f.Close()

	reader := csv.NewReader(f)
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	var people []Person
	refDate := time.Date(2025, 7, 1, 0, 0, 0, 0, time.UTC)

	for i, record := range records {
		if i == 0 || len(record) == 0 || strings.TrimSpace(record[0]) == "" {
			continue
		}

		name := record[0]
		bdayStr := record[1]
		father := record[3]
		mother := record[4]
		brother := record[5]
		sister := record[6]

		// Extract first and last name from full name
		parts := strings.Fields(name)
		if len(parts) == 0 {
			continue
		}
		fName := parts[0]
		lName := parts[len(parts)-1]

		// Parse birthday (MM/DD/YYYY)
		bday, err := time.Parse("1/2/2006", bdayStr)
		if err != nil {
			fmt.Fprintf(os.Stderr, "Error parsing date %s: %v\n", bdayStr, err)
			continue
		}

		// Calculate age as of July 1, 2025
		age := refDate.Year() - bday.Year()
		if refDate.Month() < bday.Month() || (refDate.Month() == bday.Month() && refDate.Day() < bday.Day()) {
			age--
		}

		birthdayFormatted := bday.Format("2006-01-02")

		// Build relatives list
		var relatives []Relative
		addRel := func(n, rel string) {
			if n != "" && n != "null" {
				p := strings.Fields(n)
				if len(p) > 0 {
					relatives = append(relatives, Relative{
						FirstName:    p[0],
						LastName:     p[len(p)-1],
						Relationship: rel,
					})
				}
			}
		}

		addRel(father, "Father")
		addRel(mother, "Mother")
		addRel(brother, "Brother")
		addRel(sister, "Sister")

		people = append(people, Person{
			FirstName: fName,
			LastName:  lName,
			Birthday:  birthdayFormatted,
			Age:       age,
			Relatives: relatives,
		})
	}

	out, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}
	fmt.Println(string(out))
}